server/errors/encoded_errors.go

```diff
@@ -75,22 +81,28 @@ func NewEncodedError(errorCode int32, message string) EncodedError {
 	}
 
 	var httpStatusCode int
+	var grpcStatusCode codes.Code
 	var code string
 	if errorCode >= cFirstAuthenticationErrorCode && errorCode < cFirstValidationErrorCode {
 		httpStatusCode = http.StatusUnauthorized
 		code = openfgapb.AuthErrorCode(errorCode).String()
+		grpcStatusCode = codes.Unauthenticated
```

Comment:
Our error handling must change. This is crazy that we go to a non-canonicalized error code and then back to a canonicalized error code. We way overcomplicated this behavior. Our protocol is grpc, so we should return grpc error status, and we can customize the error messages appropriately.

Let's put this on the backlog to revisit the error code strategy in favor of a more canonicalized grpc error status methodology.